iac-extensions/arm/src/main/java/org/sonar/iac/arm/tree/impl/bicep/expression/BinaryExpressionImpl.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.tree.impl.bicep.expression;
+
+import org.sonar.iac.arm.tree.api.Expression;
+import org.sonar.iac.arm.tree.api.bicep.SyntaxToken;
+import org.sonar.iac.arm.tree.api.bicep.expression.BinaryExpression;
+import org.sonar.iac.common.api.tree.SeparatedList;
+
+public class BinaryExpressionImpl extends BinaryOperationExpressionImpl implements BinaryExpression {
```

Comment:
After thinking about it, IMO I would do the following:

1. Remove BinaryOperationExpression 
2. Rename BinaryOperationExpressionImpl to AbstractBinaryOperation

I don't see that much of a value of the BinaryOperationExpression interface.

WDYT?

Reply:
I agree and tried to remove the interface, but it is needed to make other interface extends from it.
Since we are mostly working with interfaces, we cannot remove it otherwise we won't have access to the `SeparatedList`.
I still kept the renaming which is much better.